src/basic/string-util.c

```diff
@@ -843,26 +811,29 @@ char* strip_tab_ansi(char **ibuf, size_t *_isz, size_t highlight[2]) {
 }
 
 char* strextend_with_separator_internal(char **x, const char *separator, ...) {
+        _cleanup_free_ char *buffer = NULL;
         size_t f, l, l_separator;
         bool need_separator;
         char *nr, *p;
         va_list ap;
 
-        assert(x);
+        if (!x)
+                x = &buffer;
 
         l = f = strlen_ptr(*x);
 
         need_separator = !isempty(*x);
         l_separator = strlen_ptr(separator);
 
         va_start(ap, separator);
-        for (;;) {
-                const char *t;
+        for (const char *t;;) {
                 size_t n;
 
                 t = va_arg(ap, const char *);
                 if (!t)
                         break;
+                if (t == POINTER_MAX)
```

Comment:
strv_extend_many continues on NULL and uses POINTER_MAX as sentinel. I think that makes more sense generally

Reply:
i agree, but that's a major rework, as there's plenty of code that assumes that if you insert NULL into an strv_new() argument list it ends the list. it's currently all over the place, hence.

also using POINTER_MAX as sentinel has this ugliness that gcc's sentinel attribute only knows NULL as a concept. i wish it was more configurable...

i am kinda tempted to say that we should probably standardize on NULL as "terminate" and POINTER_MAX as "ignore" across the codebase for cases like this.

Reply:
Let's say it like this. i think it would be fine to normalize that either way, but let's do that separately and comprehensively, since we should probably also drop the STRV_IGNORE define then.

Some one would have to look at all invocations of strv_new(), strv_extend_many(), strextend(), and related calls, verify that they are not using an explicit NULL as a terminator somewhere, and then switch things over.

Reply:
(setting green label again hence, hope that's ok)